### PR TITLE
feat(mobile): embutir cliente expo-updates + procedência de binário (051-A PR 1.6)

### PR DESCRIPTION
**Pull request by coelhotv**
Habilita OTA no mobile (spec 051-A · FR-001..005, 007, 016 · ADR-082/083). Fatia **binária**: runbook, governança, rollout gradual e CI (051-B) ficam fora.

## O que entra

**Config do cliente** — `expo-updates` 29.0.19 (dist-tag `sdk-54`; `latest` é 57.x e quebraria o SDK 54), `runtimeVersion` literal, canal por perfil, perfil `preview` novo.

**Identidade do bundle** — `platform/updates/bundleInfo.ts` como fonte única para tela de Perfil, Sentry (tags via `initialScope`) e PostHog (super properties).

**Procedência do binário (R-307, regra nova)** — `lib-release-tag.sh` + gates no `publish-ota.sh`. Nasceu de uma dor do PO: *"como refazer o build 2803 para testar algo?"* Não havia resposta — 28 tags no repo, nenhuma de build mobile.

## Três decisões que valem revisão

**1. `runtimeVersion` é literal, não `{ policy: 'appVersion' }`.** O `expo prebuild` que antecede `eas build --local` põe o projeto em *bare workflow*, onde políticas abortam o build.

**2. O canal é declarado em DOIS lugares.** `eas.json` roteia no servidor; `app.config.js` `updates.requestHeaders` é o que fica **gravado no binário**. O prebuild regenera o manifesto a partir do `app.config.js` e apaga o que o `eas update:configure` escreveu. Pego no smoke: sem isso o app funcionaria normalmente e **nunca atualizaria** — falha que só apareceria depois de publicado na loja.

**3. Code signing DESATIVADO** (emenda ao ADR-083). O EAS cobra assinatura de update no plano **Enterprise**. A premissa do ADR era factualmente falsa; descoberta no primeiro `eas update`. Chaves preservadas; defesa do canal passa a ser 2FA + publish manual + zero `EXPO_TOKEN` em CI. PO-SEC-1 fica sem objeto.

## Bug que o smoke pegou e nenhum gate pegaria

`Updates.updateId` **não é null** em bundle embutido. O label ramificava por ele, então "Versão original" era código morto e a tela afirmava *"Atualização 25d83045"* a quem rodava o binário original — mentira sobre a única informação que a tela existe para dar. Corrigido para ramificar por `isEmbeddedLaunch`, com teste de regressão.

## Smoke em device real (Android)

| | |
|---|---|
| PO-1 update aplica | ✅ |
| PO-2 rollback | ✅ bug antigo reapareceu = prova de código antigo |
| PO-5 anti-bricking | ✅ crash proposital degrada 1 abertura, recupera sozinho na 3ª |
| PO-6 identidade na tela | ✅ |
| T017 boot offline | ✅ + achado: "2 launches" exige rede boa **no instante do boot** |
| PO-SEC-1 | sem objeto |
| T017d isolamento de canal | por inspeção (verificado no APK via `aapt2`) — decisão do PO |

## Fora de escopo, registrado

Preferência de estoque ressuscita offline (fail-safe do AP-277 não distingue "nunca soube" de "usuário desligou"). É spec 044 → **slice W1.7** próprio.

## Gates

2204 testes (164 arquivos) · lint 0 erros · `tsc` web 0 · strict-island ratchet OK · 528 testes mobile

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/mobile/src/features/profile/screens/ProfileScreen.tsx [hunk 1]
```diff
@@ -272,11 +280,23 @@ export default function ProfileScreen() {
           <Text style={styles.versionText}>
             Versão {Constants.expoConfig?.version || '0.0.0'} (Build {Constants.expoConfig?.ios?.buildNumber || Constants.expoConfig?.android?.versionCode || '1'})
           </Text>
+          {/* FR-016/PO-6: com OTA, "versão 0.30.0" deixa de identificar o código — dois aparelhos
+              na mesma versão podem rodar bundles diferentes. Esta linha é o que o suporte pede ao
+              usuário pra saber QUAL bundle está na mão dele. Selecionável pra caber num print/cópia. */}
+          <Text style={styles.bundleText} selectable numberOfLines={1}>
+            {BUNDLE_LABEL}
+          </Text>
+          {CHANNEL_LABEL && (
```
Comment (coelhotv) [inline apps/mobile/src/features/profile/screens/ProfileScreen.tsx:289]: 🟨 **medium** [checklist#6]

Unsafe logical AND with a string variable in React Native JSX. If CHANNEL_LABEL evaluates to an empty string, it yields "" inside a View, causing a React Native rendering crash.

**Causation:** If formatChannelLabel returns an empty string ("") when no update channel is present, evaluating CHANNEL_LABEL && <Text> in JSX yields "" as a direct child of <View style={styles.versionSection}> (ProfileScreen.tsx:L289). React Native throws a runtime crash ("Text strings must be rendered within a <Text> component") when primitive string values are evaluated directly inside a View without being wrapped in Text or converted to a boolean.

**Fix:** Use explicit boolean coercion or a ternary check: Boolean(CHANNEL_LABEL) && (...) or CHANNEL_LABEL ? (...) : null

**Review by coelhotv — COMMENTED** [review]
## 🤖 RC6 — Independent AI Review (`agy (2 chunks)`)

- **pass A · chunk 2:** The PR introduces bundle identity tracking for Expo OTA updates (bundleInfo.ts), attaches bundle tags to Sentry crashes and PostHog analytics in AppRoot.tsx, and adds a network error utility (networkError.ts) to improve offline user messaging. All added modules include comprehensive unit test coverage, follow fail-silent telemetry guidelines, and properly handle edge cases such as empty channel strings or missing OTA updates.
- **pass B · chunk 1:** The PR bumps the mobile version to 0.30.0, configures EAS Update / OTA update settings in app.config.js, integrates bundle/channel metadata display on ProfileScreen.tsx, and improves network error handling via describeError in useProfile.ts. The audit identified an unsafe logical AND expression with a string variable in React Native JSX inside ProfileScreen.tsx that can evaluate to an empty string child inside a View component, triggering a runtime crash in React Native.
- **pass B · chunk 2:** Audit completed for mobile OTA bundle info and telemetry updates. One introduced finding was identified in bundleInfo.ts where nullish coalescing fails to handle native empty string channel values, sending empty tags to telemetries instead of the mandated fallback. All other changes in AppRoot.tsx and networkError.ts were verified as safe and contract-compliant.

**Coverage:** 2/2 chunks reviewed
**Introduced:** 0 critical · 0 high (total 0c/0h/2m/0l). Findings marked _pre-existing_ do not block this PR.

File: apps/mobile/src/platform/updates/bundleInfo.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * Nunca omite chave: ausência de OTA vira o valor literal `embedded`, não `undefined`. Tag que
+ * some quando o valor é nulo cria o pior caso de investigação — "os crashes sem update_id" vira
+ * um balde onde não dá pra distinguir "rodava o embutido" de "a instrumentação falhou".
+ */
+export function bundleTags(): Record<string, string> {
+  const info = getBundleInfo()
+  return {
+    update_id: info.updateId ?? 'embedded',
+    channel: info.channel ?? 'none',
```
Comment (coelhotv) [inline apps/mobile/src/platform/updates/bundleInfo.ts]: 🟨 **medium** [checklist#7]

Nullish coalescing operator (??) does not catch empty string channel values returned by native updates, emitting an empty tag to Sentry initialScope and PostHog super properties when no OTA channel is configured.

**Causation:** The native expo-updates module returns an empty string '' when channel is missing (as documented in formatChannelLabel and verified in bundleInfo.test.ts). In JavaScript, '' ?? 'none' evaluates to '' because '' is not nullish (null or undefined), causing bundleTags() to return an empty string channel tag instead of 'none'.

**Fix:** channel: info.channel?.trim() || 'none',

**Comment by coelhotv** [comment]
## Resposta ao RC6

**`bundleInfo.ts:67` — `??` não pega string vazia · CONFIRMADO, corrigido em `03163a1`**

Achado correto e revelador: a mesma string vazia do nativo que eu havia tratado no caminho da **tela** (`formatChannelLabel`, com `trim()`) seguia intacta no caminho da **telemetria**. O comentário logo acima da função declara querer evitar exatamente o balde onde "rodava o embutido" e "a instrumentação falhou" viram a mesma coisa — e o código emitia tag vazia, que produz esse balde.

Corrigido nos três campos, não só no canal. E a causa de fundo: **`bundleTags` não tinha teste algum**. Foram adicionados 4 (campos preenchidos, ausência total, string vazia do nativo, canal só-de-espaços).

---

**`ProfileScreen.tsx:289` — `CHANNEL_LABEL &&` renderizando `""` · FALSO POSITIVO**

A premissa não se sustenta: `formatChannelLabel` tem retorno `string | null` e nunca devolve `""` —

```ts
export function formatChannelLabel(info: BundleInfo): string | null {
  const canal = info.channel?.trim()
  return canal ? `Canal ${canal}` : null
}
```

Canal vazio ou só-de-espaços vira `null`, e `null && <Text>` não renderiza nada. Coberto por três testes (`''`, `'   '`, `null`).

O padrão apontado é real em geral — só não se aplica aqui, porque a função foi escrita justamente para não devolver string vazia (regressão do separador órfão achado no smoke). Sem mudança.

---

Gates após o fix: 532 testes mobile · lint 0 erros.

File: apps/mobile/src/features/profile/screens/ProfileScreen.tsx (hunk 1 above)
Comment (coelhotv) [reply apps/mobile/src/features/profile/screens/ProfileScreen.tsx:289]: **Falso positivo** — a premissa não se sustenta.

`formatChannelLabel` tem retorno `string | null` e nunca devolve `""`:

```ts
export function formatChannelLabel(info: BundleInfo): string | null {
  const canal = info.channel?.trim()
  return canal ? `Canal ${canal}` : null
}
```

Canal vazio, só-de-espaços ou `null` viram `null`, e `null && <Text>` não renderiza nada. Coberto por três testes nomeados justamente por isso (`bundleInfo.test.ts`) — a função foi escrita para não devolver string vazia, porque o separador órfão que ela evita foi um bug real do smoke.

O padrão apontado é legítimo em geral; aqui não se aplica. Sem mudança.

File: apps/mobile/src/platform/updates/bundleInfo.ts (hunk 2 above)
Comment (coelhotv) [reply apps/mobile/src/platform/updates/bundleInfo.ts]: **Confirmado e corrigido** em `03163a1`.

Achado certeiro. Eu já conhecia esse comportamento do nativo — tinha corrigido a mesma string vazia em `formatChannelLabel` horas antes, no smoke, inclusive escrevendo um comentário explicando que o nativo devolve `""` e não `null`. O que não fiz foi propagar para o **segundo consumidor do mesmo fato**, 30 linhas acima no mesmo arquivo. Correção pela metade.

Aplicado nos três campos, não só no canal (`update_id` e `runtime_version` tinham a mesma exposição):

```ts
const clean = (value: string | null | undefined, fallback: string): string =>
  value?.trim() || fallback
```

E a causa de fundo, que importa mais que a linha: **`bundleTags` não tinha teste algum**. Foi por aí que passou. Adicionados 4 — campos preenchidos, ausência total, string vazia do nativo, canal só-de-espaços.

Registrado como medição 034-D.3 (spec 034): o gate de `introduced critical/high` não teria disparado neste caso, e o achado merecia correção mesmo assim — severity de finding de observabilidade é subestimada por natureza, porque o dano não aparece no fluxo, aparece na próxima investigação.
